taipy/gui/utils/_evaluator.py

```diff
@@ -293,15 +295,23 @@ def re_evaluate_expr(self, gui: Gui, var_name: str) -> t.Set[str]:
             expr_original = self.__hash_to_expr[var_name]
             temp_expr_var_map = self.__expr_to_var_map[expr_original]
             if len(temp_expr_var_map) <= 1:
+                index_in_array = int(m[0]) if (m := _Evaluator.__IS_ARRAY_EXPR_RE.findall(expr_original)) else -1
                 # since this is an edge case --> only 1 item in the dict and that item is the original var
-                for v in temp_expr_var_map.values():
-                    var_name = v
+                var_name = next(iter(temp_expr_var_map.values()), var_name)
                 # construct correct var_path to reassign values
                 var_name_full, _ = _variable_decode(expr_original)
                 var_name_full = var_name_full.split(".")
                 var_name_full[0] = var_name
                 var_name_full = ".".join(var_name_full)
-                _setscopeattr_drill(gui, var_name_full, _getscopeattr(gui, var_name_original))
+                if index_in_array > - 1:
```

Comment:
I think it would be better using >= 0

Reply:
I kind of agree with @namnguyen20999... >= 0 is toe me more intuitive than > -1.... it's not just me

Reply:
everybody agrees even though it's 1 character less :-)